File: 21) Kth Distinct String in an Array/code.py
n, k = map(int, input("Enter n, k: ").split())
arr = []
mp = {}

print("Enter strings:")
for _ in range(n):
    string = input()
    arr.append(string)
    mp[string] = mp.get(string, 0) + 1

ans = ""
for string in arr:
    if mp[string] == 1:
        if k == 1:
            ans = string
            break
        else:
            k -= 1

if ans == "":
    print("ans:")
else:
    print("ans:", ans)

File: 21) Kth Distinct String in an Array/code.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int n, k;
    cout << "Enter n, k: ";
    cin >> n >> k;
    vector<string> arr(n);
    map<string, int> mp;
    cout << "Enter strings: ";
    for(int i=0; i<n; i++) {
        cin >> arr[i];
        mp[arr[i]]++;
    }
    
    string ans;
    for(int i=0; i<n; i++) {
        if(mp[arr[i]] == 1) {
            if(k==1) {
                ans = arr[i];
                break;
            }
            else k--;
        }
    }
    
    if(ans.length() == 0) {
        cout << "ans: " << endl;
    } else {
        cout << "ans: " << ans << endl;
    }
    
    return 0;
}